src/librustdoc/visit_ast.rs

```diff
@@ -329,25 +329,21 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
         if !self.view_item_stack.insert(def_node_id) { return false }
 
         let ret = match tcx.hir.get(def_node_id) {
-            hir_map::NodeItem(it) => {
+            hir_map::NodeItem(&hir::Item { node: hir::ItemMod(ref m), .. }) if glob => {
                 let prev = mem::replace(&mut self.inlining, true);
-                if glob {
-                    match it.node {
-                        hir::ItemMod(ref m) => {
-                            for i in &m.item_ids {
-                                let i = self.cx.tcx.hir.expect_item(i.id);
-                                self.visit_item(i, None, om);
-                            }
-                        }
-                        hir::ItemEnum(..) => {}
-                        _ => { panic!("glob not mapped to a module or enum"); }
```

Comment:
It took me a couple reads to understand what was going on here, but just to straighten out my understanding: Reworking the match like this means that glob imports of enums will never be inlined, making them always appear as `pub use SomeEnum::*;`, whereas before they would be *treated* as they were inlined, but the variants would never be visited, leaving them completely undocumented? I haven't worked much with the AST crawling portion of rustdoc, but looking at the comments at the head of the function and the other change in `clean/inline.rs`, that's the impression I get from this change.